corehq/apps/users/tests/test_role_utils.py

```diff
@@ -118,3 +125,75 @@ def _delete_presets(self):
         for role in UserRole.objects.get_by_domain(self.domain):
             if role.id != self.role1.id:
                 role.delete()
+
+
+class TestCommcareAnalyticsRolesByUser(TestCase):
+
+    DOMAIN = "domain1"
+    USERNAME = "username"
+    PASSWORD = "***"
+
+    @classmethod
+    def setUpClass(cls):
+        cls.domain = create_domain(cls.DOMAIN)
+        cls.user = WebUser.create("domain1", cls.USERNAME, cls.PASSWORD, None, None)
```

Comment:
(Not blocking.) I've started using `self.addCleanup()` and `cls.addClassCleanup()` instead of defining teardown methods, because if anything in your setup fails, it will still tear down the things that were created. I've found it can avoid annoying issues with conflicting domain and usernames while writing tests.

```suggestion
        cls.domain = create_domain(cls.DOMAIN)
        cls.addClassCleanup(cls.domain.delete)
        cls.user = WebUser.create("domain1", cls.USERNAME, cls.PASSWORD, None, None)
        cls.addClassCleanup(cls.user.delete, cls.DOMAIN, None)
        # ... etc. ...
```

Reply:
> instead of defining teardown methods

Just confirming that you meant for data created in class setup only right? and not for data created in individual tests because that gets cleaned up after every test.
So, just `addClassCleanup` and not  `addCleanUp`?